File: 1_1.cpp
#include <iostream>

int gcd(int a, int b, int &x, int &y)
{
  if (a == 0)
  {
    x = 0;
    y = 1;
    return b;
  }

  int x1, y1;
  int d = gcd(b % a, a, x1, y1);
  std::cout << "x=" << x1 << " y=" << y1 << "\n";
  x = y1 - (b / a) * x1;
  y = x1;
  return d;
}

int main()
{
  int x, y;
  gcd(546, 212, x, y);
  std::cout << "x=" << x << " y=" << y << " " << 546 * x + 212 * y << "\n";
  return 0;
}

File: 1_1.py
def rem(x, y):
    q = 0
    r = x
    while y <= r:
        r = r - y
        q += 1
    return q, r


def gcd(m, n):
    p, r = rem(m, n)
    init_m = m
    init_n = n

    parts = [p]
    steps = 1

    while r > 0:
        m = n
        n = r
        p, r = rem(m, n)
        parts.append(p)
        steps += 1

    a = 0
    b = 1
    for i in range(steps-1, -1, -1):
        _a = a
        a = b - parts[i] * a
        b = _a

    print(b*init_m + a*init_n, n)
    return n


print(gcd(546, 212))
